stl/inc/xmemory

```diff
@@ -1033,147 +1034,194 @@ struct _Iterator_base0 {
-#endif // _ITERATOR_DEBUG_LEVEL == 2
+        } else { // no future parent, just disown current parent
+            _Orphan_me_v2();
         }
     }
 
-    const _Container_base12* _Getcont() const noexcept {
-        return _Myproxy ? _Myproxy->_Mycont : nullptr;
-    }
-
-#if _ITERATOR_DEBUG_LEVEL == 2
-    void _Orphan_me() noexcept {
+    _CONSTEXPR20_CONTAINER void _Orphan_me_v2() noexcept {
         if (_Myproxy) { // adopted, remove self from list
-            _Iterator_base12** _Pnext = &_Myproxy->_Myfirstiter;
-            while (*_Pnext && *_Pnext != this) {
-                _Pnext = &(*_Pnext)->_Mynextiter;
+#ifdef __cpp_lib_constexpr_dynamic_alloc
+            if (_STD is_constant_evaluated()) {
+                _Orphan_me_unlocked();
+            } else
+#endif // __cpp_lib_constexpr_dynamic_alloc
+            {
+                _Orphan_me_locked();
             }
+        }
+    }
 
-            _STL_VERIFY(*_Pnext, "ITERATOR LIST CORRUPTED!");
-            *_Pnext  = _Mynextiter;
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
+    _CONSTEXPR20_CONTAINER void _Adopt(const _Container_base12* _Parent) noexcept {
+        if (_Parent) { // have a parent, do adoption
+            _Myproxy = _Parent->_Myproxy;
+        } else { // no future parent, just disown current parent
             _Myproxy = nullptr;
         }
     }
-#endif // _ITERATOR_DEBUG_LEVEL == 2
+#endif // _ITERATOR_DEBUG_LEVEL != 2
+
+    _CONSTEXPR20_CONTAINER const _Container_base12* _Getcont() const noexcept {
+        return _Myproxy ? _Myproxy->_Mycont : nullptr;
+    }
 
     static constexpr bool _Unwrap_when_unverified = _ITERATOR_DEBUG_LEVEL == 0;
 
-    _Container_proxy* _Myproxy;
-    _Iterator_base12* _Mynextiter;
-};
+    _Container_proxy* _Myproxy    = nullptr;
+    _Iterator_base12* _Mynextiter = nullptr;
 
-// MEMBER FUNCTIONS FOR _Container_base12
-inline void _Container_base12::_Orphan_all() noexcept {
 #if _ITERATOR_DEBUG_LEVEL == 2
-    if (_Myproxy) { // proxy allocated, drain it
+private:
+    _CONSTEXPR20_CONTAINER void _Adopt_unlocked(_Container_proxy* _Parent_proxy) noexcept {
+        if (_Myproxy) { // adopted, remove self from list
+            _Orphan_me_unlocked();
+        }
+        _Mynextiter                 = _Parent_proxy->_Myfirstiter;
+        _Parent_proxy->_Myfirstiter = this;
+        _Myproxy                    = _Parent_proxy;
+    }
+
+    void _Adopt_locked(_Container_proxy* _Parent_proxy) noexcept {
         _Lockit _Lock(_LOCK_DEBUG);
+        _Adopt_unlocked(_Parent_proxy);
+    }
 
-        for (auto _Pnext = &_Myproxy->_Myfirstiter; *_Pnext; *_Pnext = (*_Pnext)->_Mynextiter) {
-            (*_Pnext)->_Myproxy = nullptr;
+    _CONSTEXPR20_CONTAINER void _Orphan_me_unlocked() noexcept {
+        _Iterator_base12** _Pnext = &_Myproxy->_Myfirstiter;
+        while (*_Pnext && *_Pnext != this) {
+            const auto _Temp = *_Pnext; // TRANSITION: ASO-XXXXX

```

Comment:
We should update this with the actual bug number (and I think you meant `ADO` 😛):
```suggestion
            const auto _Temp = *_Pnext; // TRANSITION: ADO-1269037
```